bundles/org.openhab.binding.bluetooth.dbusbluez/src/main/java/org/openhab/binding/bluetooth/dbusbluez/handler/DBusBlueZBridgeHandler.java

```diff
@@ -0,0 +1,343 @@
+            return false;
+        }
+
+        try {
+            Thread.sleep(100);
+        } catch (InterruptedException e1) {
+            Thread.currentThread().interrupt();
+        }
+
+        // This call will return for sure the DM, otherwize createInstance
+        // would have failed previously
+        this.deviceManager = DeviceManager.getInstance();
+
+        if (this.deviceManager != null) {
+            logger.debug("Device Manager correctly instanciated");
+
+            // a handler must be instanciated to get all notifications
+            // from DBUS (new device, RSSI update, characteristic notification...)
+            try {
+                this.deviceManager.registerPropertyHandler(this.propertiesChangedHandler);
+            } catch (DBusException e) {
+                // Shoudl not happen..
+                logger.error("Error registering properties changed handler", e);
+                return false;
+            }
+        } else {
+            // should normally not happen..
+            logger.debug("Device Manager could not be instanciated but no error.");
+        }
+
+        return (this.deviceManager != null);
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+
+    private void initializeDeviceManagerInternal() throws DBusException {
+        try {
+            // if this is the first call to the library, this call
+            // should throw an exception (that we are catching)
+            this.deviceManager = DeviceManager.getInstance();
+
+            // Experimental - seems reuse does not work
+            this.deviceManager.closeConnection();
+            DeviceManager.createInstance(false);
+        } catch (IllegalStateException e) {
+            // Exception caused by first call to the library
+            DeviceManager.createInstance(false);
+        }
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+
+    /**
+     * This function finds the adapter providing the address in param
+     *
+     * @return
+     */
+    private boolean initializeAdapter() {
+
+        List<com.github.hypfvieh.bluetooth.wrapper.BluetoothAdapter> adapters = this.deviceManager.getAdapters();
+
+        if (adapters == null || adapters.isEmpty()) {
+            return false;
+        }
+
+        for (com.github.hypfvieh.bluetooth.wrapper.BluetoothAdapter btAdapter : adapters) {
+            if (btAdapter.getAddress() != null
+                    && btAdapter.getAddress().equalsIgnoreCase(this.adapterAddress.toString())) {
+                // Found the good adapter
+                this.adapter = btAdapter;
+
+                logger.debug("Turning off adapter...");
+                // Power cycle OFF / ON for a clean start
+                this.adapter.setPowered(false);
+                logger.debug("Adapter state: {}", this.adapter.isPowered());
+
+                DBusBlueZUtils.sleep(1000);
```

Comment:
Instead of sleeping you should schedule a delayed task using `scheduler`. Bindings should make initialization as fast as possible since it is executed from a UI thread. Any time consuming operation during initialization should be done in a background thread.